irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -74,17 +74,17 @@ namespace iroha {
       auto wsv_transaction =
           std::make_unique<pqxx::nontransaction>(*postgres_connection, kTmpWsv);
 
-      boost::optional<shared_model::interface::types::HashType> top_hash;
-
-      getBlockQuery()
-          ->getTopBlocks(1)
-          .subscribe_on(rxcpp::observe_on_new_thread())
-          .as_blocking()
-          .subscribe([&top_hash](auto block) { top_hash = block->hash(); });
-
+      auto block_result = getBlockQuery()->getTopBlock();
       return expected::makeValue<std::unique_ptr<MutableStorage>>(
           std::make_unique<MutableStorageImpl>(
-              top_hash.value_or(shared_model::interface::types::HashType("")),
+              block_result.match(
+                  [](expected::Value<
+                      std::shared_ptr<shared_model::interface::Block>> block) {
+                    return block.value->hash();
+                  },
+                  [](expected::Error<std::string> error) {
```

Comment:
You can omit the name of the parameter if it is unused. Also pass by reference to avoid extra copies